repose-aggregator/tests/functional-tests/src/integrationTest/groovy/features/services/uriredaction/UriRedactionServiceKeystoneTest.groovy

```diff
@@ -0,0 +1,207 @@
+/*
+ * _=_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_=
+ * Repose
+ * _-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-
+ * Copyright (C) 2010 - 2015 Rackspace US, Inc.
+ * _-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * =_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_=_
+ */
+package features.services.uriredaction
+
+import org.openrepose.framework.test.ReposeValveTest
+import org.openrepose.framework.test.mocks.MockIdentityV2Service
+import org.openrepose.framework.test.mocks.MockTracerCollector
+import org.rackspace.deproxy.Deproxy
+import org.rackspace.deproxy.MessageChain
+
+class UriRedactionServiceKeystoneTest extends ReposeValveTest {
```

Comment:
I would love a test case capturing the behavior when capture text appears more than once in a URI. So like, using the already defined regex `/[a-zA-Z]+/specific/(path)`, if the URI is `/path/specific/(path)`, I would expect the redacted URI to be `/path/specific/XXXXX` and not `/XXXXX/specific/XXXXX`.

Reply:
Added this during the `Unroll` flattening.